packages/utils/src/conversions.ts

```diff
@@ -0,0 +1,202 @@
+/**
+ * @module Conversions
+ * This module provides a collection of utility functions for converting
+ * between different numerical formats, particularly focusing on
+ * conversions involving bigints, hexadecimals and buffers.
+ * The module is structured with clear function naming to indicate
+ * the conversion direction (e.g., `bigIntToHexadecimal` for BigInt
+ * to hexadecimal, `bufferToBigint` for buffer to bigint) and employs
+ * type checks to ensure the correct handling of various input types.
+ * It also includes variations for both big-endian (`be`) and little-endian
+ * (`le`) conversions. It is important to note that when there is no prefix,
+ * the order of bytes is always big-endian.
+ */
+
+import { BigNumberish } from "./types"
+import { isHexadecimal, isStringifiedBigint } from "./number-checks"
+
+/**
+ * Converts a bigint to a hexadecimal string ensuring even length.
+ * @param n The bigint value to convert.
+ * @returns The hexadecimal representation of the bigint.
+ */
+export function beBigIntToHexadecimal(n: bigint): string {
+    let hex = n.toString(16)
+
+    // Ensure even length.
+    if (hex.length % 2 !== 0) {
+        hex = `0${hex}`
+    }
+
+    return hex
+}
+
+/**
+ * Converts a bigint to a hexadecimal string. Alias for beBigIntToHexadecimal.
+ * @param n The bigint value to convert.
+ * @returns The hexadecimal representation of the bigint.
+ */
+export function bigIntToHexadecimal(n: bigint): string {
+    return beBigIntToHexadecimal(n)
+}
+
+/**
+ * Converts a hexadecimal string to a bigint, ensuring the hex string starts with '0x'.
+ * @param hex The hexadecimal string to convert.
+ * @returns The bigint representation of the hexadecimal string.
+ */
+export function beHexadecimalToBigint(hex: string): bigint {
+    // Ensure the hex string starts with '0x'.
+    const formattedHexString = hex.startsWith("0x") ? hex : `0x${hex}`
+
+    return BigInt(formattedHexString)
+}
+
+/**
+ * Converts a hexadecimal string to a bigint. Alias for beHexadecimalToBigint.
+ * @param hex The hexadecimal string to convert.
+ * @returns The bigint representation of the hexadecimal string.
+ */
+export function hexadecimalToBigint(hex: string): bigint {
+    return beHexadecimalToBigint(hex)
+}
+
+/**
+ * Converts a buffer to a bigint by interpreting the buffer as a hexadecimal string.
```

Comment:
"by interpreting the buffer as a hexadecimal string" is an implementation detail which I think is actually confusing here.  It suggests to me that as caller I should be passing an array of strings.  The buffer isn't being intereted as a hex string, it's being converted to a hex string internally via the call to toString() below.

Alternative suggestion: "Converts a buffer of bytes to a bigint using big-endian byte order."